### ISSUE TITLE
Reduce bundle size

### ISSUE THREAD

**Issue by fumblehool** [securingsincity/react-ace#282]
I analyzed by `bundle.js` using `webpack-bundle-analyzer` and It is nearly half the size of total bundle. I've tried different releases of react-ace but all have same result 
![screen shot 2017-10-05 at 5 52 26 pm](https://user-images.githubusercontent.com/8335453/31227088-3cb96c8c-a9f6-11e7-97cf-23ce0b294717.png)

How can I reduce the bundle size?

**Comment by securingsincity** [comment]
unfortunately @fumblehool brace and ace are enormous libraries. I would love to find a way to reduce the bundle size as well.

**Comment by fumblehool** [comment]
Is there some way to do this? I would love to help you.

**Comment by securingsincity** [comment]
I would be open to any help. I'm not sure what it is that is driving the size other than the dependencies.

**Comment by NSLS** [comment]
You could split your bundle on top level routes which use react-ace.
For example, if you have a route /posts/new for a page, which includes the editor, you could load the page's component asynchronously with System.import.

In react-router 3 this is possible by using getComponent prop on <Route>

This won't solve the problem of the huge size of ace and brace, and I think that nothing would, every wysiwyg library is huge in size. But at least your website won't load this extra 1 mb of data when it is not needed

**Comment by jan-osch** [comment]
Brace makes up to 66% of my relatively complex SPA build with webpack (create-react-app). 

Do you know any way to reduce the size of `brace`? Is there any aggressive minification you can apply? It would really help

**Comment by MarvanN** [comment]
Looking forward for good answer to this.

**Comment by jondavidjohn** [comment]
I built a tool to help you analyze webpack bundles for size regressions, and report them directly to GitHub PRs. It's free for open source, so it might be worth checking out and helpful in this scenario.

[https://packtracker.io](https://packtracker.io/?utm_campaign=links&utm_source=github&utm_medium=issueprcomment)

![image](https://user-images.githubusercontent.com/401520/58719965-8ae35c80-8396-11e9-908d-5c5b1d7b8cbf.png)

**Comment by securingsincity** [comment]
so with v8 of react-ace we've dropped support for brace and only should be importing the ace files  you absolutely need. In theory bundle sizes should be smaller but i haven't tested this.

**Comment by TeoTN** [comment]
Not really, it's even a bit worse as brace was 640.83KB whereas ace-builds is 665.48KB.
It's huge :/

**Comment by rongyadong** [comment]
any idea that we can reduce the bundle size?

**Comment by Srinjoy-Santra** [comment]
I would so like to have the bundle size reduced. Currently, I am using [code splitting](https://reactjs.org/docs/code-splitting.html) to lazy load the editor component.

**Comment by travisdahl** [comment]
Running into the same problem where the bundle size is unacceptable for my project. I was trying `react-syntax-highlighter` because they have a _light_ version. May be worth looking into how they handle it. I think the ace plugin is actually better and was hoping it had a similar light version.

**Comment by ashishRudder** [comment]
I also need a lighter version. Having an `sql` editor with `textmate` theme and auto-completion is more than enough for my work. Don't need any other fancy features in that.

**Comment by Bonfims** [comment]
> I also need a lighter version. Having an `sql` editor with `textmate` theme and auto-completion is more than enough for my work. Don't need any other fancy features in that.

indeed, i just want a functional json editor to a very small and simple project, and all other features that i will not use worries me.

Awesome work by the way
